dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreCollectionCreateMapping.cs

```diff
@@ -48,11 +48,13 @@ internal static class RedisVectorStoreCollectionCreateMapping
     /// </summary>
     /// <param name="properties">The property definitions to map from.</param>
     /// <param name="storagePropertyNames">A dictionary that maps from a property name to the storage name that should be used when serializing it to json for data and vector properties.</param>
+    /// <param name="useDollarPrefix">A value indicating whether to include $. prefix for field names as required in JSON mode.</param>
     /// <returns>The mapped Redis <see cref="Schema"/>.</returns>
     /// <exception cref="InvalidOperationException">Thrown if there are missing required or unsupported configuration options set.</exception>
-    public static Schema MapToSchema(IEnumerable<VectorStoreRecordProperty> properties, Dictionary<string, string> storagePropertyNames)
+    public static Schema MapToSchema(IEnumerable<VectorStoreRecordProperty> properties, Dictionary<string, string> storagePropertyNames, bool useDollarPrefix)
```

Comment:
Should instead of a bool be something more open ended?

```suggestion
    public static Schema MapToSchema(IEnumerable<VectorStoreRecordProperty> properties, Dictionary<string, string> storagePropertyNames, string? fieldNamePrefix)
```

Reply:
There are only two options, $. for JSON or nothing for HASHES.  It's not that you can pass in anything and it'll work.  So having a bool is nice because it exposes the choice but doesn't require the caller to know more about the implementation.